quiz/src/Category.js

```diff
@@ -0,0 +1,35 @@
+import React ,{useState,useEffect} from 'react';
+import {CategoryData,DifficultyData} from './Data';
+
+function Category () {
+  const[cat,setCat]=useState('');
+  const[diff,setDiff]=useState('');
+  const [questions,setQuestions]=useState([]);
+  useEffect(()=>{
+    fetch(`https://opentdb.com/api.php?amount=10&category=${cat}&difficulty=${diff}`)
+    .then(res => res.json()).then(({results})=>setQuestions(results))
+  })
```

Comment:
**Good work at using `useEffect` :+1:**  ,
 - but you should let it run only  when the `diff` or `cat` changes.
 - To detect what is happening try :` console.log(results)` , you will find that results will be 
    consoled in an infinite loop without changing the value of `diff` and `cat`.
**SO,** 
`You can tell React to skip applying an effect if certain values (cat, diff) haven’t changed between re-renders. To do so, pass an array as an optional second argument to useEffect: `